### ISSUE TITLE
Traverse directories while skipping dotfiles

### ISSUE THREAD

**Issue by budziq** [brson/rust-cookbook#202]
Please use [`walkdir`](https://docs.rs/walkdir) crate to show how we can operate on a directory recursively while efficiently omitting hidden files and directories ("dotfiles").

The idea is to show off usage of [`filter_entry`](https://docs.rs/walkdir/*/walkdir/trait.WalkDirIterator.html#method.filter_entry) 

.

**Comment by tyoc213** [comment]
HI there, I have basically this

```rust

fn should_skip(entry: &DirEntry) -> bool {
    !entry.file_name()
         .to_str()
         .map(|s| s != "." && s.starts_with("."))
         .unwrap()
}

fn run() -> Result<()> {
    for entry in WalkDir::new(".").into_iter().filter_entry(|e| should_skip(e)) {
        println!("{:?}", entry);
    }
    Ok(())
}
```

The only problem Im having is that if I mark the current directory as starting with "." or "hidden" it inmediately stops, because being a directory and returning that it should be filtered out, the whole listing of the directory and subdirs is gone :).

So it is OK like that? or should I skip always the first entry being the current directory ".".

**Comment by budziq** [comment]
Hi, nice :+1:  you can always use [`DirEntry::depth`](https://docs.rs/walkdir/*/walkdir/struct.DirEntry.html#method.depth) to never skip the starting dir.

Also would suggest to always start with a draft PR (even if its almost empty). It is easier to give guidance (and further action items). I have few minor suggestions so far:

- do not `unwrap` or `expect`. `should_skip` can use `unwrap_or` for a safe default on failure
- please print out the path of entry with normal display trait "{}"

**Comment by budziq** [comment]
closed with #203
